src/coreclr/src/vm/gcinfodecoder.cpp

```diff
@@ -1686,7 +1686,17 @@ OBJECTREF* GcInfoDecoder::GetCapturedRegister(
     PREGDISPLAY     pRD
     )
 {
-    _ASSERTE(regNum >= 0 && regNum <= 28);
+    _ASSERTE(regNum >= 0 && regNum <= 30);
```

Comment:
So I am not sure here....
In the debugger, the `pRD->pCurrentContextPointers->Fp` was null so get stack slot forwarded here.  The code didn't support that case, so the assert made sense.

In the case I was observing, the `pRD->pCurrentContext->Fp` looked valid, so it seemed reasonable and harmless to support it here.  

However, this did not fix any of the tests hitting this assert....  So perhaps another fix is needed...  This still seemed functionally correct to me, so I committed it anyway (at least to trigger discussion).  If you know why we should never have to support this case, it might help me find the true root cause of the issue....

Reply:
Root cause.  For MacOS `HAVE_UNW_GET_SAVE_LOC` is not defined.  This is because macos libunwind.h has this.
```C++
//extern int       unw_get_save_loc(unw_cursor_t*, int, unw_save_loc_t*);
```